tools/advisor/advisor/db_config_optimizer.py

```diff
@@ -0,0 +1,385 @@
+        # remove the ambiguous guidelines
+        for option in options_to_remove:
+            final_guidelines.pop(option, None)
+        return final_guidelines
+
+    def get_guidelines(self, rules, suggestions_dict):
+        guidelines = {}
+        for rule in rules:
+            new_scope = rule.get_trigger_column_families()
+            for sugg_name in rule.get_suggestions():
+                option = suggestions_dict[sugg_name].option
+                action = suggestions_dict[sugg_name].action
+                if not (option or action):  # suggestion not in required format
+                    continue
+                sugg_values = suggestions_dict[sugg_name].suggested_values
+                if not sugg_values:
+                    sugg_values = []
+                if option not in guidelines:
+                    guidelines[option] = self.get_guideline_boiler_plate()
+                guidelines[option][action][self.SCOPE] = (
+                    guidelines[option][action][self.SCOPE].union(new_scope)
+                )
+                guidelines[option][action][self.SUGG_VAL] = (
+                    guidelines[option][action][self.SUGG_VAL].union(
+                        sugg_values
+                    )
+                )
+        guidelines = self.disambiguate_guidelines(guidelines)
+        return guidelines
+
+    def run(self, num_iterations):
+        new_options = copy.deepcopy(self.db_options)
+        for _ in range(num_iterations):
+            # run the benchmarking tool for the given database configuration
+            print('\nbenchrunner experiment: ')
+            data_sources, _ = self.bench_runner.run_experiment(new_options)
+            # load and run the rule parser with the obtained data sources
+            self.rule_parser.load_rules_from_spec()
+            self.rule_parser.perform_section_checks()
+            triggered_rules = self.rule_parser.get_triggered_rules(
+                data_sources, new_options.get_column_families()
+            )
+            if not triggered_rules:
+                print('No rules triggered, exiting optimizer!')
+                break
+            print('\nTriggered:')
+            self.rule_parser.print_rules(triggered_rules)
+            # convert triggered rules to optimizer understandable advisor
+            # guidelines
+            guidelines = self.get_guidelines(
+                triggered_rules, self.rule_parser.get_suggestions_dict()
+            )
+            # use the guidelines to improve the database configuration
+            working_config = new_options.get_options(list(guidelines.keys()))
+            updated_config = ConfigOptimizer.improve_db_config(
+                working_config, guidelines
+            )
+            print('current config:\n' + repr(working_config))
+            print('db config changes:\n' + repr(updated_config))
+            new_options.update_options(updated_config)
+        return new_options
+
+    def run_v2(self):
```

Comment:
Can you add an inline comment of what run_v2 is doing and how is it different than run?